src/main/java/de/zalando/ep/zalenium/registry/ZaleniumRegistry.java

```diff
@@ -449,6 +450,31 @@ public void uncaughtException(Thread t, Throwable e) {
         }
     }
 
+    @Override
+    public HttpClient getHttpClient(URL url) {
+        // https://github.com/zalando/zalenium/issues/491
+        int maxTries = 3;
+        for (int i = 0; i < maxTries; i++) {
+            try {
+                HttpClient client = httpClientFactory.createClient(url);
+                if (i > 0) {
+                    String message = String.format("Successfully created HttpClient for url %s, after attempt #%s", url, (i+1));
+                    LOG.warn(message);
+                }
+                return client;
+            } catch (Exception | AssertionError e) {
+                String message = String.format("Error while getting the HttpClient for url %s, attempt #%s", url, (i+1));
+                LOG.debug(message, e);
+                if (i == 2) {
+                    throw e;
+                }
+            }
+        }
+        LOG.warn(String.format("Last attempt to get the HttpClient for url %s", url));
```

Comment:
Will this log every time a client is requested? It looks this way?

Reply:
Oh my bad. There is a return statement there. I read it too quickly.

Reply:
I’m not sure if that will log at all will it? Since it’s after the throw and outside the for loop

Reply:
Yes, that is dead code. I'll remove it, I need to leave the return though.